packages/types/src/plugin.ts

```diff
@@ -43,3 +45,18 @@ export interface Plugin<PluginContext extends Record<string, any> = {}> {
    */
   onContextBuilding?: OnContextBuildingHook<PluginContext>;
 }
+
+export type Tracer<TContext extends Record<string, any>> = {
```

Comment:
So the Tracer manages the lifecycle of the whole req? I wonder if the tracer needs more core info or that's something that's up to the end-user (or, gw) to decide in their implementation?

Reply:
My idea was to provide the very minimal info there. If you need more infos, you have the hooks.
I give access to the context (and later to the request in Yoga, and the subgraph execution request in Hive Gateway) to have a stable reference you can use as a key in a weakmap for this purpose.

For example, in OTEL, I would start the span in the tracer, and then add relevant attributes in the hooks. Hooks will have access to the span either by using OTEL context manager, or via a weakmap using the context as key.